scripts/experimental/configure_for_workload_identity.sh

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env bash
+
+# Prior to running this script, please ensure that you are running these versions or later:
+# export SPINNAKER_VERSION=release-1.17.x-latest-validated
+# export HALYARD_VERSION=1.26.0
+#
+# This script is intended to be run after the initial setup.sh script completes and Spinnaker is up
+# and running (without Workload Identity enabled).
+#
+# The expected workflow is as follows:
+#   - Generate the properties file by running the setup_properties.sh script
+#   - Modify the properties file to specify the above 2 Spinnaker/Halyard versions (or later versions)
+#   - Run setup.sh
+#   - Once Spinnaker is up and running, run this (configure_for_workload_identity.sh) script
+#
+# Note that this script results in each Spinnaker pod still using the default Kubernetes service account,
+#   and the default service account in the halyard and spinnaker namespaces being bound to one Google
+#   service account (spinnaker-wi-acct). If you want to specify a different Kubernetes service account
+#   for any service, you can do so via the `serviceAccountName` setting described here:
+#   https://www.spinnaker.io/reference/halyard/custom/#kubernetes
+#   You would also need to make the appropriate bindings between that Kubernetes service account and a
+#   Google service account.
+#
+#   The roles assigned are sufficient for deployment to GKE. If you intend to deploy to GCE or GAE, you
+#     will need to assign the appropriate roles to the spinnaker-wi-acct Google service account, similar
+#     to what we do in these helper scripts for the non-Workload Identity setup:
+#       https://github.com/GoogleCloudPlatform/spinnaker-for-gcp/blob/master/scripts/manage/add_gce_account.sh
+#       https://github.com/GoogleCloudPlatform/spinnaker-for-gcp/blob/master/scripts/manage/add_gae_account.sh
+
+bold() {
+  echo ". $(tput bold)" "$*" "$(tput sgr0)";
+}
+
+PROPERTIES_FILE="$HOME/spinnaker-for-gcp/scripts/install/properties"
+
+source "$PROPERTIES_FILE"
+
+bold "Scaling down deployments to 0..."
+kubectl scale deployment spin-clouddriver -n spinnaker --replicas=0
```

Comment:
Why are we scaling these deployments down to 0 before we enable workload identity, then back up after?  The documentation on migrating workloads seems to imply that updating the node pool is enough.

Reply:
The issue I ran into was that the pods are all rescheduled once the node pool is updated. This resulted in auth failures, and they never seemed to recover from this (it's possible this relates to not-yet-expired tokens. but I got tired of waiting).

I'm glad you asked about this as it forced me to work through it again. By moving the node pool update to the end, after the service accounts have been bound and the roles assigned, the auth failures are avoided and the scaling can be dropped. Thanks.